src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -391,7 +393,12 @@ public void Serialize(XmlWriter xmlWriter, object? o, XmlSerializerNamespaces? n
                     }
                 }
                 else
-                    _tempAssembly.InvokeWriter(_mapping, xmlWriter, o, namespaces == null || namespaces.Count == 0 ? DefaultNamespaces : namespaces, encodingStyle, id);
+                {
+                    using (AssemblyLoadContext.EnterContextualReflection(_tempAssembly._assembly))
+                    {
```

Comment:
Why was this needed here? I wouldn't expect any code generation on this call path as the assembly has already been loaded/created.

Reply:
We can discuss in our weekly meeting, but if we're executing a read/write on a serializer that is loaded in a specific context, wouldn't we want any by/products of that execution to also live within that context?